File: ios/Riskified.m
#import "React/RCTBridgeModule.h"

@interface RCT_EXTERN_MODULE(Riskified, NSObject)
RCT_EXTERN_METHOD(initRiskified:(NSString *)MerchantId userToken:(NSString *)userToken)
RCT_EXTERN_METHOD(logRequest:(NSString *)path)
RCT_EXTERN_METHOD(updateSessionToken:(NSString *)newToken)
@end

File: ios/Riskified.swift
import Foundation

@objc(Riskified)
class Riskified: NSObject {
  
  @objc
  func initRiskified(_ MerchantId: String, userToken: String) {
    if(RiskifiedGlobalState.isInitialized) {
      return
    }
    
    RiskifiedBeacon.start(MerchantId, sessionToken: userToken, debugInfo: false)
    RiskifiedGlobalState.isInitialized = true
  }
  
  @objc
  func logRequest(_ path: String) {
    if(!RiskifiedGlobalState.isInitialized) {
      return
    }
    
    RiskifiedBeacon.logRequest(URL(string: path))
  }
  
  @objc
  func updateSessionToken(_ newToken: String) {
    if(!RiskifiedGlobalState.isInitialized) {
      return
    }
    
    RiskifiedBeacon.updateSessionToken(newToken)
  }
  
  @objc
   static func requiresMainQueueSetup() -> Bool {
     return true
   }
}
